lib/optimizer/src/Error/ErrorProperties.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+namespace Amp\Optimizer\Error;
+
+/**
+ * Default set of properties and methods to use for errors.
+ *
+ * @package Amp\Optimizer
+ */
+trait ErrorProperties
+{
+
+    /**
+     * Instantiate an Error object.
+     *
+     * @param string $message Message for the error.
+     */
+    public function __construct($message)
+    {
+        $this->message = $message;
+    }
+
+    /**
+     * Message of the error.
+     *
+     * @var string
+     */
+    protected $message;
+
+    /**
+     * Get the code of the error.
+     *
+     * @return string Code of the error.
+     */
+    public function getCode()
+    {
+        return basename(get_class($this));
```

Comment:
This doesn't seem to be working as expected. I'm seeing `Amp\Optimizer\Error\CannotInlineRuntimeCss`.

```suggestion
        return preg_replace('/^.+\\\\/', '', get_class($this));
```